packages/react-dom/src/events/__tests__/DOMModernPluginEventSystem-test.internal.js

```diff
@@ -1074,5 +1074,527 @@ describe('DOMModernPluginEventSystem', () => {
+            <div ref={divRef}>Click me!</div>
+          </button>
+        );
+      }
+
+      ReactDOM.render(<Test />, container);
+      Scheduler.unstable_flushAll();
+
+      let buttonElement = buttonRef.current;
+      dispatchClickEvent(buttonElement);
+      expect(onClick).toHaveBeenCalledTimes(1);
+      expect(onClickCapture).toHaveBeenCalledTimes(1);
+      expect(log[0]).toEqual(['capture', buttonElement]);
+      expect(log[1]).toEqual(['bubble', buttonElement]);
+
+      let divElement = divRef.current;
+      dispatchClickEvent(divElement);
+      expect(onClick).toHaveBeenCalledTimes(3);
+      expect(onClickCapture).toHaveBeenCalledTimes(3);
+      expect(log[2]).toEqual(['capture', buttonElement]);
+      expect(log[3]).toEqual(['capture', divElement]);
+      expect(log[4]).toEqual(['bubble', divElement]);
+      expect(log[5]).toEqual(['bubble', buttonElement]);
+    });
+
+    it('should correctly work for a basic "click" listener on the outer target', () => {
+      const log = [];
+      const clickEvent = jest.fn(event => {
+        log.push({
+          eventPhase: event.eventPhase,
+          type: event.type,
+          currentTarget: event.currentTarget,
+          target: event.target,
+        });
+      });
+      const divRef = React.createRef();
+      const buttonRef = React.createRef();
+
+      function Test() {
+        const click = ReactDOM.unstable_useEvent('click');
+
+        React.useEffect(() => {
+          click.setListener(divRef.current, clickEvent);
+        });
+
+        return (
+          <button ref={buttonRef}>
+            <div ref={divRef}>Click me!</div>
+          </button>
+        );
+      }
+
+      ReactDOM.render(<Test />, container);
+      Scheduler.unstable_flushAll();
+
+      expect(container.innerHTML).toBe('<button><div>Click me!</div></button>');
+
+      // Clicking the button should trigger the event callback
+      let divElement = divRef.current;
+      dispatchClickEvent(divElement);
+      expect(log[0]).toEqual({
+        eventPhase: 3,
+        type: 'click',
+        currentTarget: divRef.current,
+        target: divRef.current,
+      });
+
+      // Unmounting the container and clicking should not work
+      ReactDOM.render(null, container);
+      dispatchClickEvent(divElement);
+      expect(clickEvent).toBeCalledTimes(1);
+
+      // Re-rendering the container and clicking should work
+      ReactDOM.render(<Test />, container);
+      Scheduler.unstable_flushAll();
+
+      divElement = divRef.current;
+      dispatchClickEvent(divElement);
+      expect(clickEvent).toBeCalledTimes(2);
+
+      // Clicking the button should not work
+      let buttonElement = buttonRef.current;
+      dispatchClickEvent(buttonElement);
+      expect(clickEvent).toBeCalledTimes(2);
+    });
+
+    it('should correctly handle many nested target listeners', () => {
+      const buttonRef = React.createRef();
+      const targetListerner1 = jest.fn();
```

Comment:
```suggestion
      const targetListener1 = jest.fn();
```
(and the ones after this one too)